sdk/easm/defender-easm-rest/samples-dev/discoTemplateSample.ts

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * @summary Create discovery groups using a template
+ *
+ * @description This sample show you how to use the easm defender client to create discovery groups using templates
+ *
+ * Set the following environment variables before running the sample:
+    1) SUBSCRIPTIONID - the subscription id for your resource
+    2) WORKSPACENAME - the workspace name for your resource
+    3) RESOURCEGROUPNAME - the resource group for your resource
+    4) REGION - the azure region your resource is in
+    5) PARTIAL_NAME - the search term for the templates. used for a case insensitive "contains" search
+ */
+
+import EasmDefender, { isUnexpected } from "@azure-rest/defender-easm";
+import { DefaultAzureCredential } from "@azure/identity";
+// Load the .env file if it exists
+import * as dotenv from "dotenv";
+dotenv.config();
+
+async function main() {
+  // To create an EasmClient, you need your subscription ID, region, and some sort of credential.
+  const subscription_id = process.env.SUBSCRIPTIONID || "";
+  const resource_group = process.env.RESOURCEGROUPNAME || "";
+  const workspace_name = process.env.WORKSPACENAME || "";
+  const region = process.env.REGION || "";
+  const endpoint = `https://${region}.easm.defender.microsoft.com`;
+  const credential = new DefaultAzureCredential();
+
+  const partial_name = process.env.PARTIAL_NAME || "";
+  console.log(`Partial name is ${partial_name}`);
+
+  const client = EasmDefender(
```

Comment:
do we usally call these methods `createXXXXXClient` by conventions, @joheredi?

Reply:
@jeremymeng, from other examples, we use constructor to create clients